app/test_main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+import pytest
+from unittest import mock
+from unittest.mock import MagicMock
+
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url,mocked_result_connection,mocked_result_url,result",
+    [
+        ("http://google.com", True, True, "Accessible"),
+        ("http://google.com", False, False, "Not accessible"),
+        ("http://google.com", True, False, "Not accessible"),
+        ("http://google.com", False, True, "Not accessible"),
+    ],
+    ids=[
+        "return 'Accessible' connection is working, url is valid",
+        "return 'Not accessible' connection is not working, url do not valid",
+        "return 'Not accessible' connection is working, url do not valid",
+        "return 'Not accessible' connection is not working, url is valid"
```

Comment:
The test names in the 'ids' list should be more descriptive and grammatically correct. For example, 'return 'Accessible' when connection is working and URL is valid' would be clearer.